include/deal.II/matrix_free/portable_matrix_free.templates.h

```diff
@@ -132,22 +138,17 @@ namespace Portable
     void
     ReinitHelper<dim, Number>::resize(const unsigned int n_colors)
     {
-      // We need at least three colors when we are using device-aware MPI and
-      // overlapping the communication
-      data->n_cells.resize(std::max(n_colors, 3U), 0);
-      data->local_to_global.resize(n_colors);
-      data->constraint_mask.resize(n_colors);
+      typename MatrixFree<dim, Number>::PerDoFHandlerData &dof_data =
+        data->dof_handler_data[dof_handler_index];
 
-      data->row_start.resize(n_colors);
-
-      if (update_flags & update_quadrature_points)
-        data->q_points.resize(n_colors);
```

Comment:
Why did you remove this line?

Reply:
I moved some stuff around inside ``internal_reinit``: Anything we need to do for each DoFHandler is in the big loop at the bottom, while other stuff that needs to be done once (row_starts, graph setup etc.) is no higher up.